tests/test_interpolation.py

```diff
@@ -211,71 +211,203 @@ def test_type_inherit_type(cfg: Any) -> None:
     assert type(cfg.s) == str  # check that string interpolations are always strings
 
 
+@pytest.mark.parametrize("env_func", ["env", "oc.env"])
+class TestEnvInterpolation:
+    @pytest.mark.parametrize(
+        "cfg,env_name,env_val,key,expected",
+        [
+            pytest.param(
+                {"path": "/test/${${env_func}:foo}"},
+                "foo",
+                "1234",
+                "path",
+                "/test/1234",
+                id="simple",
+            ),
+            pytest.param(
+                {"path": "/test/${${env_func}:not_found,ZZZ}"},
+                None,
+                None,
+                "path",
+                "/test/ZZZ",
+                id="not_found_with_default",
+            ),
+            pytest.param(
+                {"path": "/test/${${env_func}:not_found,a/b}"},
+                None,
+                None,
+                "path",
+                "/test/a/b",
+                id="not_found_with_default",
+            ),
+        ],
+    )
+    def test_env_interpolation(
+        self,
+        monkeypatch: Any,
+        env_func: str,
+        cfg: Any,
+        env_name: Optional[str],
+        env_val: str,
+        key: str,
+        expected: Any,
+    ) -> None:
+        if env_name is not None:
+            monkeypatch.setenv(env_name, env_val)
+
+        cfg["env_func"] = env_func  # allows choosing which env resolver to use
+        cfg = OmegaConf.create(cfg)
+
+        # The legacy env resolver triggers a deprecation warning.
+        if env_func == "env":
+            with pytest.warns(UserWarning):
+                assert OmegaConf.select(cfg, key) == expected
+        else:
+            assert OmegaConf.select(cfg, key) == expected
+
+    @pytest.mark.parametrize(
+        ("cfg", "key", "expected"),
+        [
+            pytest.param(
+                {"path": "/test/${${env_func}:not_found}"},
+                "path",
+                pytest.raises(
+                    InterpolationResolutionError,
+                    match=re.escape("Environment variable 'not_found' not found"),
+                ),
+                id="not_found",
+            ),
+        ],
+    )
+    def test_env_interpolation_error(
+        self,
+        env_func: str,
+        cfg: Any,
+        key: str,
+        expected: Any,
+    ) -> None:
+        cfg["env_func"] = env_func  # allows choosing which env resolver to use
+        cfg = _ensure_container(cfg)
+
+        # The legacy env resolver triggers a deprecation warning.
```

Comment:
ditto. add one test for directly testing the deprecation warning and ignore the warnings everywhere else.

Reply:
side note, we should probably split test_interpolation into something like test_simple_interpolations.py and test_custom_resolvers.py.

Reply:
> ditto. add one test for directly testing the deprecation warning and ignore the warnings everywhere else.

Done in a973d21 (I didn't add a new test since there are a couple of tests specific to the legacy env that still explicitly catch that warning, doesn't seem worth doing more refactoring since they will all go away in 2.2)

Reply:
> side note, we should probably split test_interpolation into something like test_simple_interpolations.py and test_custom_resolvers.py.

Noted, will do in a follow-up PR